### PR TITLE
Remove extra copying properties

### PR DESCRIPTION
**Pull request by kokosing**

### CODE REVIEW

File: presto-sqlserver/src/test/java/io/prestosql/plugin/sqlserver/SqlServerQueryRunner.java [hunk 1]
```diff
@@ -60,7 +60,7 @@ public static QueryRunner createSqlServerQueryRunner(TestingSqlServer testingSql
             queryRunner.installPlugin(new TpchPlugin());
             queryRunner.createCatalog("tpch", "tpch");
 
-            connectorProperties = new HashMap<>(ImmutableMap.copyOf(connectorProperties));
```
Comment (findepi) [inline presto-sqlserver/src/test/java/io/prestosql/plugin/sqlserver/SqlServerQueryRunner.java]: it was verifying there are no nulls

**Comment by electrum** [comment]
Commit title nit
> Remove extra copying of properties

File: presto-sqlserver/src/test/java/io/prestosql/plugin/sqlserver/SqlServerQueryRunner.java (hunk 1 above)
Comment (electrum) [reply presto-sqlserver/src/test/java/io/prestosql/plugin/sqlserver/SqlServerQueryRunner.java]: I had the same thought initially, but assumed it was leftover from a refactoring. In the case that it's intentional, let's add a comment.

Comment (martint) [reply presto-sqlserver/src/test/java/io/prestosql/plugin/sqlserver/SqlServerQueryRunner.java]: Or make the verification explicit:

```java
checkArgument(!map.containsKey(null) && !map.containsValue(null), ...);
```

**Comment by findepi** [comment]
We use this pattern in some other places already.
eg
- `new ArrayList<>(ImmutableList.copyOf(` ...
- `new HashMap<>(ImmutableMap.copyOf(` ...

I used to think the intent is readable and obvious.
If it's not obvious, let's add a comment.

I don't like `checkArgument(!map.containsKey(null) && !map.containsValue(null), ...);` option as overly verbose.
(Might be good for production code, as it allows for a better exc msg, but too verbose for test code.)

**Comment by kokosing** [comment]
> let's add a comment.

Instead I would create a dedicated method for that and replace other places like that.

**Comment by findepi** [comment]
> Instead I would create a dedicated method for that and replace other places like that.

For oneliner, this could be an overkill. And there is no shared module other than SPI, so...

**Comment by martint** [comment]
> option as overly verbose.

A comment is not going to make it much less verbose than the checkArgument version. Plus, it has the downside that the java compiler can't check the comment to make sure it's still accurate ;)

**Comment by findepi** [comment]
> Why should tests be less readable than production code? ;)

Tests should be readable. But they don't need to produce as good exc msgs as prod code.



> null-intolerance is almost a hidden behavior of `copyOf`.

Isn't that obvious at least now?

Look, we leverage Immutable's "maintain ordering" behavior, which is totally invisible, non-obvious and counter-intuitive.

This one is intuitive (once you think about it).

What about leaving the code as-is? Is this an option?

**Comment by martint** [comment]
> Tests should be readable. But they don't need to produce as good exc msgs as prod code.

You can remove the message out of the checkArgument. That's not the important part.

> Isn't that obvious at least now?

Only to people very familiar with Guava. It's not obvious from the name of the method, and the way it's being used makes it appear as if we're "just trying to copy it but left screwed up the code in some refactoring" (as mentioned above).

It's important for the intent to be as clear as possible in the code. If you have to explain the intent, then something's wrong with the code or the problem being solved is too complex and additional context is needed. (That's why names are important, for instance).

**Comment by kokosing** [comment]
Updated.

**Pull request closed without merging**